src/systemdhook.c

```diff
@@ -570,25 +512,32 @@ static int prestart(const char *rootfs,
 		}
 	}
 
-	if (!contains_mount(config_mounts, config_mounts_len, "/sys/fs/cgroup")) {
-		/* libmount */
-		struct libmnt_table *tb = NULL;
-		int rc = -1;
+	if (!contains_mount(config_mounts, config_mounts_len, CGROUP_SYSTEMD)) {
+		_cleanup_free_ char *dest = NULL;
+		rc = asprintf(&dest, "%s/%s", rootfs, CGROUP_SYSTEMD);
 
-		/*
-		 * initialize libmount
-		 */
-		mnt_init_debug(0);
+		if ((umount(dest) == -1) && errno != EINVAL) {
+			pr_perror("Failed to umount %s", dest);
+		}
 
-		tb = parse_tabfile("/proc/self/mountinfo");
-		if (!tb) {
+		if (mount("cgroup", dest, "cgroup", MS_NOSUID|MS_NOEXEC|MS_NODEV, "none,name=systemd,xattr") == -1) {
+			pr_perror("Failed to mount systemd %s", dest);
 			return -1;
 		}
 
-		rc = mount_cgroup(tb, NULL, rootfs);
-		mnt_free_table(tb);
-		if (rc == -1) {
-			return -1;
+		if (strcmp("", mount_label)) {
+			if (setfilecon(dest, mount_label) < 0) {
```

Comment:
Is this section failing or just reporting that it can't set the labels. I had hoped the kernel would return ENOTSUP and we could keep it in place for when cgroup gains support.